metricbeat/module/mongodb/replstatus/replstatus.go

```diff
@@ -0,0 +1,85 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package replstatus
+
+import (
+	mgo "gopkg.in/mgo.v2"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/module/mongodb"
+)
+
+var debugf = logp.MakeDebug("mongodb.replstatus")
+
+func init() {
+	mb.Registry.MustAddMetricSet("mongodb", "replstatus", New,
+		mb.WithHostParser(mongodb.ParseURL),
+		mb.DefaultMetricSet())
```

Comment:
If this requires replicaset to work maybe it'd be better to make it non-default.